src/main/java/com/venafi/vcert/sdk/connectors/tpp/TppTokenConnector.java

```diff
@@ -51,76 +57,93 @@ public String getVendorAndProductName() {
         return vendorAndProductName;
     }
 
-    private String getAuthHeaderValue(String token) throws VCertException {
-        if(isBlank(token)){
+    private String getAuthHeaderValue() throws VCertException {
+        if(isBlank(credentials.accessToken())){
             throw new VCertException("Token cannot be empty");
         }
 
-        return String.format(HEADER_VALUE_AUTHORIZATION, token);
+        return String.format(HEADER_VALUE_AUTHORIZATION, credentials.accessToken());
     }
 
     @Override
-    public void ping(String accessToken) throws VCertException {
-        Response response = doPing(accessToken);
+    public void ping() throws VCertException {
+        Response response = doPing();
         if (response.status() != 200) {
             throw new VCertException(
                     format("ping failed with status %d and reason %s", response.status(), response.reason()));
         }
     }
 
-    private Response doPing(String accessToken) throws VCertException{
-        return tpp.pingToken(getAuthHeaderValue(accessToken));
+    private Response doPing() throws VCertException{
+        return tpp.pingToken(getAuthHeaderValue());
     }
 
     @Override
     public TokenInfo getAccessToken(Authentication auth) throws VCertException {
-
-        VCertException.throwIfNull( auth, MISSING_CREDENTIALS_MESSAGE );
+        if(isEmptyCredentials(auth)) {
+            throw new VCertException(MISSING_CREDENTIALS_MESSAGE);
+        }
 
         AuthorizeTokenRequest info = new AuthorizeTokenRequest( auth.user(), auth.password(), auth.clientId(), auth.scope(), auth.state(), auth.redirectUri() );
```

Comment:
Not important comment: I would not call it 'info' as it is some sort of request

Reply:
Variable has been renamed with a better, self-explaining name.